odxtools/diaglayer.py

```diff
@@ -454,49 +455,115 @@ def get_communication_parameter(self, name: str) \
 
         return cps[0]
 
+    def _get_comparam_value(self,
+                            com_param : CommunicationParameterRef) \
+            -> Optional[str]:
+        """Retrieve the value of a simple communication parameter
+
+        This takes the default value of the comparam (if any) into
+        account.
+        """
+
+        assert isinstance(com_param.comparam, Comparam)
+
+        result: Any = None
+        if com_param.value:
+            result = com_param.value
+        else:
+            result = com_param.comparam.physical_default_value
+
+        assert isinstance(result, str)
+
+        return result
+
+    def _get_comparam_subvalue(self,
```

Comment:
What happens when a complex param also contains a complex?

Reply:
bad luck ;)

(I think for the standardized communication parameters this is never the case?)